DTS-STN/Service-Canada-Labs
# Dynamic Article Generation

# [Implement first set of components for article generator](https://dev.azure.com/VP-BD/DECD/_boards/board/t/Service%20Canada%20Labs/Stories%20and%20Activities)

This PR includes changes to support the dynamic generation of article pages based on fragment data received from AEM.

The first implementation of this is with the Benefits Navigator dynamic article page. The page loops through the fragments, and renders components that correspond to those fragments. Content editors can rearrange the order of elements on the page and of course modify their content. The fragments supported in article pages are currently:

TextWithImage
![Screenshot 2023-12-05 at 2 00 57 PM](https://github.com/DTS-STN/Service-Canada-Labs/assets/31868510/b58fec13-d658-4fae-a27f-7d6955acc248)

Buttons
![Screenshot 2023-12-05 at 2 01 31 PM](https://github.com/DTS-STN/Service-Canada-Labs/assets/31868510/2bfc1b33-5218-436f-8a10-131ad39e4ea5)

TextContent
![Screenshot 2023-12-05 at 2 01 56 PM](https://github.com/DTS-STN/Service-Canada-Labs/assets/31868510/5a1b4725-f637-4d1f-8e64-6c9682bcfb26)


## Test Instructions

1. Navigate to Benefits Navigator project page
2. Navigate to Test article
3. See that the page renders without crashing (some fields like Posted on: and Last updated: are empty)
4. With Postman, in the Benefits Navigator Article Pages response, see that the order of fragments in the scFragments property of the test article reflect the order of elements on the page
